lib/views/internallogin_screen.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:gocast_mobile/main.dart';
+
+class InternalloginScreen extends ConsumerWidget {
+  const InternalloginScreen({super.key});
+
+  Future<void> handleBasicLogin(
+    BuildContext context,
+    WidgetRef ref,
+    TextEditingController usernameController,
+    TextEditingController passwordController,
+  ) async {
+    // Call the basic authentication function from /base/api/auth
+    await ref
+        .read(userViewModel)
+        .basicAuth(usernameController.text, passwordController.text)
+        .then(
+          (value) => {
+            if (ref.read(userViewModel).current.value.user != null)
+              {Navigator.pushNamed(context, '/welcome')},
+          },
+        );
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final TextEditingController usernameController = TextEditingController();
+    final TextEditingController passwordController = TextEditingController();
+
+    return Scaffold(
```

Comment:
Extract some code into function or widgets, the function body is quite long

Reply:
This will be done in this [branche](https://github.com/TUM-Dev/gocast_mobile/tree/34-clean-code)